### PR TITLE
CR-1071356 Integrate verify, bandwidth and slave bridge git examples in xbutil validate

### PR DESCRIPTION
**Pull request by AShivangi**
- Cases handled:
```
            |     Old XRT       |    New XRT
Old shell   |   Runs python     |  Runs python
New shell   |  Throws an error  |  Runs cpp
```
- Versal will always run py test case since we don't have the cpp equivalent
- Todo: add to xbutil2

### CODE REVIEW

File: src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp [hunk 1]
```diff
@@ -1186,37 +1186,70 @@ int xcldev::device::runTestCase(const std::string& py,
...
+    // if platform.json exists in the new shell pakcage then use the new testcase
+    // else fallback to old python test cases
+    
+    // NEW FLOW: runs if platform.json is available and the platform is not versal
+    // Note: we don't have cpp based versal bw testcase
+    auto json_exists = [xclbinPath]() 
+        {
+            return boost::filesystem::exists(xclbinPath + "platform.json") ? true : false;
+        };
+    if(json_exists() && xclbin.compare("versal_23_bandwidth.py") != 0) {
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp:1206]: We can have platform.json for versal platform?

File: src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp [hunk 2]
```diff
@@ -1186,37 +1186,70 @@ int xcldev::device::runTestCase(const std::string& py,
...
+        
+        //map old testcase names to new testcase names
+        static const std::map<std::string, std::string> test_map = {
+            { "22_verify.py",             "validate.exe"    },
+            { "23_bandwidth.py",          "kernel_bw.exe"   },
+            { "host_mem_23_bandwidth.py", "slavebridge.exe" }
+        };
+        
+        xrtTestCasePath += test_map.find(py)->second;
+        // in case the user is trying to run a new platform with an older XRT
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp:1216]: Could you clarify "older XRT“？

**Review by maxzhen — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp (hunk 1 above)
Comment (AShivangi) [reply src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp:1206]: I don't think Heera's team has bw testcase for versal yet. We'll have to rely on python testcase until they add the versal bw

File: src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp (hunk 2 above)
Comment (AShivangi) [reply src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp:1216]: By older XRT I mean any XRT before this PR, but I'll update the comment reflecting this

File: src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp (hunk 1 above)
Comment (maxzhen) [reply src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp:1206]: do you mean that they can have verify kernel for versal, but no bandwidth for versal?

File: src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp (hunk 2 above)
Comment (maxzhen) [reply src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp]: > 
> 
> By older XRT I mean any XRT before this PR, but I'll update the comment reflecting this

you pr will make "older XRT" become "new XRT", so how do you handle "older XRT" case in your pr?

Comment (AShivangi) [reply src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp]: So if we have a new shell (with platform.json) with an older XRT (without the .exe), then we'll throw an error saying: 
`ERROR: Failed to find verify.exe. Please update XRT.`

File: src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp (hunk 1 above)
Comment (AShivangi) [reply src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp:1206]: Oh, I guess I'm mistaken. I don't think that verify kernel for versal works either. We should check and see if they are providing platform.json for versal. If not then  `xclbin.compare("versal_23_bandwidth.py") != 0` is redundant

File: src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp [hunk 3]
```diff
@@ -1186,37 +1186,67 @@ int xcldev::device::runTestCase(const std::string& py,
...
-        output += ", Shell package not installed properly.";
-        return -ENOENT;
-    }
+    std::string cmd;
+    
+    // if platform.json exists in the new shell pakcage then use the new testcase
+    // else fallback to old python test cases
+    
+    // NEW FLOW: runs if platform.json is available and the platform is not versal
+    // Note: we don't have cpp based versal bw testcase
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp:1201]: I don't think you want to call out versal. there is no difference than other platforms here.

**Review by maxzhen — CHANGES_REQUESTED** [review]

**Review by maxzhen — APPROVED** [review]

**Pull request merged**